flow/activities/flowable_core.go

```diff
@@ -292,27 +291,38 @@ func (a *FlowableActivity) getPostgresPeerConfigs(ctx context.Context) ([]*proto
 }
 
 // replicateQRepPartition replicates a QRepPartition from the source to the destination.
-func (a *FlowableActivity) replicateQRepPartition(ctx context.Context,
+func replicateQRepPartition[TRead any, TWrite any, TSync connectors.QRepSyncConnectorCore, TPull connectors.QRepPullConnectorCore](
```

Comment:
function signature is a bit complex
a) use a struct to simplify some inputs
b) idx and total are only being used for heartbeat, make it caller side

Reply:
unfortunately go makes generic function type aliases not a thing. moving heartbeats out would be good. have heartbeat cleanup happening in another pr, #1742, so maybe after as a followup to that